src/FormFile.js

```diff
@@ -0,0 +1,190 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useContext, useMemo } from 'react';
+import all from 'prop-types-extra/lib/all';
+import Feedback from './Feedback';
+import FormFileInput from './FormFileInput';
+import FormFileLabel from './FormFileLabel';
+import FormContext from './FormContext';
+import { useBootstrapPrefix } from './ThemeProvider';
+
+const propTypes = {
+  /**
+   * @default 'form-file'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * A seperate bsPrefix used for custom controls
+   *
+   * @default 'custom-file'
+   */
+  bsCustomPrefix: PropTypes.string,
+
+  /**
+   * The FormFile `ref` will be forwarded to the underlying input element,
+   * which means it will be a DOM node, when resolved.
+   *
+   * @type {ReactRef}
+   * @alias ref
+   */
+  _ref: PropTypes.any,
+
+  /**
+   * The underlying HTML element to use when rendering the FormFile.
+   *
+   * @type {('input'|elementType)}
+   */
+  as: PropTypes.elementType,
+
+  /** A HTML id attribute, necessary for proper form accessibility. */
+  id: PropTypes.string,
+
+  /**
+   * Provide a function child to manually handle the layout of the FormFile's inner components.
+   *
+   * ```jsx
+   * <FormFile>
+   *   <FormFile.Input isInvalid />
+   *   <FormFile.Label>Allow us to contact you?</FormFile.Label>
+   *   <Feedback type="invalid">Yo this is required</Feedback>
+   * </FormFile>
+   * ```
+   */
+  children: PropTypes.node,
+
+  disabled: PropTypes.bool,
+  label: PropTypes.node,
+
+  /** Use Bootstrap's custom form elements to replace the browser defaults */
+  custom: PropTypes.bool,
+
+  /** Manually style the input as valid */
+  isValid: PropTypes.bool.isRequired,
+
+  /** Manually style the input as invalid */
+  isInvalid: PropTypes.bool.isRequired,
+
+  /** A message to display when the input is in a validation state */
+  feedback: PropTypes.node,
+
+  /** The string for the button when using custom file input */
+  buttonText: all(PropTypes.string, ({ custom, buttonText }) =>
```

Comment:
Been thinking about this some more after some discussion (not on GitHub) and looking at the Bootstrap docs again. Should this prop be remove and the data attribute be passed via ...props instead? This would better align with the official Bootstrap documentation.

However it would, in the case of using
```
  <Form.File
    id="custom-file-translate"
    label="Custom file input"
    data-browse="Button text"
    custom
  />
```
Also be passed to the input element and adding unneeded data attributes to that element.

Reply:
Removed buttonText prop and instead uses data-browse attribute as TWBS to keep consistency with them.